lib/queuePopulator/QueuePopulator.js

```diff
@@ -236,20 +236,21 @@ class QueuePopulator {
     }
 
     _processAllLogEntries(params, done) {
-        return async.map(
+        return async.each(
             this.logReaders,
-            (logReader, done) => logReader.processAllLogEntries(params, done),
-            (err, results) => {
-                if (err) {
-                    return done(err);
-                }
-                const annotatedResults = results.map(
-                    (result, i) => Object.assign(result, {
-                        logSource: this.logReaders[i].getLogInfo(),
-                        logOffset: this.logReaders[i].getLogOffset(),
-                    }));
-                return done(null, annotatedResults);
-            });
+            (logReader, readerDone) => {
+                const batchCb = (err, processedAll) => {
+                    if (err) {
+                        return readerDone(err);
+                    }
+                    if (processedAll || this.logReadersUpdate) {
```

Comment:
is it sufficient to know if an update is in progress ? is it the circuit breaker ?

Reply:
Yes it's the circuit breaker, because it's checked by the calling function `QueuePopulator.processAllLogEntries`